apps/studio/src/lib/db/clients/mysql.ts

```diff
@@ -134,14 +134,19 @@ async function configDatabase(
   database: IDbConnectionDatabase
 ): Promise<mysql.PoolOptions> {
 
+  let awsCLIToken = undefined;
+  if( server.config.redshiftOptions.authType === 'iam_cli') {
+    awsCLIToken = await getAWSCLIToken(server.config, server.config.redshiftOptions);
+  }
+
   const config: mysql.PoolOptions = {
     authPlugins: {
       'client_ed25519': ed25519AuthPlugin(),
     },
     host: server.config.host,
     port: server.config.port,
     user: server.config.user,
-    password: await refreshTokenIfNeeded(server.config.redshiftOptions, server, server.config.port || 3306) || server.config.password || undefined,
+    password: awsCLIToken || await refreshTokenIfNeeded(server.config.redshiftOptions, server, server.config.port || 3306) || server.config.password || undefined,
```

Comment:
This could be a problem. If we aren't in iam_cli mode, it's still going to try to refresh the token. This should be in the if statement up above, just check if the token is undefined after getting it, then refresh if needed. Then here you just do this
```suggestion
    password: awsCLIToken || server.config.password,
```